src/preferences/colorpalettesettings.cpp

```diff
@@ -0,0 +1,52 @@
+#include "preferences/colorpalettesettings.h"
+
+const QString ColorPaletteSettings::hotcueColorPaletteGroup = QStringLiteral("[HotcueColorPalette]");
+const QString ColorPaletteSettings::trackColorPaletteGroup = QStringLiteral("[TrackColorPalette]");
+
+ColorPalette ColorPaletteSettings::getColorPalette(
+        const QString& group, const ColorPalette& defaultPalette) const {
+    QList<mixxx::RgbColor> colorList;
+    for (const ConfigKey& key : m_pConfig->getKeysWithGroup(group)) {
```

Comment:
qAsConst() around collection (although probably not implicitly shared)

Reply:
```
/home/jan/Projects/mixxx/src/preferences/colorpalettesettings.cpp: In member function 'ColorPalette ColorPaletteSettings::getColorPalette(const QString&, const ColorPalette&) const':
/home/jan/Projects/mixxx/src/preferences/colorpalettesettings.cpp:9:76: error: use of deleted function 'void qAsConst(const T&&) [with T = QList<ConfigKey>]'
    9 |     for (const ConfigKey& key : qAsConst(m_pConfig->getKeysWithGroup(group))) {
      |                                                                            ^
In file included from /usr/include/qt/QtCore/qsharedpointer.h:43,
                 from /usr/include/qt/QtCore/QSharedPointer:1,
                 from /home/jan/Projects/mixxx/src/preferences/usersettings.h:4,
                 from /home/jan/Projects/mixxx/src/preferences/colorpalettesettings.h:3,
                 from /home/jan/Projects/mixxx/src/preferences/colorpalettesettings.cpp:1:
/usr/include/qt/QtCore/qglobal.h:1043:6: note: declared here
 1043 | void qAsConst(const T &&) = delete;
      |      ^~~~~~~~
make[2]: *** [CMakeFiles/mixxx-lib.dir/build.make:4002: CMakeFiles/mixxx-lib.dir/src/preferences/colorpalettesettings.cpp.o] Error 1
```